public/app/features/alerting/unified/components/rule-editor/notificaton-preview/NotificationPreviewByAlertManager.tsx

```diff
@@ -0,0 +1,113 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Alert, LoadingPlaceholder, useStyles2 } from '@grafana/ui';
+
+import { Stack } from '../../../../../../plugins/datasource/parca/QueryEditor/Stack';
+import { Labels } from '../../../../../../types/unified-alerting-dto';
+
+import { NotificationRoute } from './NotificationRoute';
+import { useAlertmanagerNotificationRoutingPreview } from './useAlertmanagerNotificationRoutingPreview';
+import { AlertManagerNameWithImage } from './useGetAlertManagersSourceNamesAndImage';
+
+function NotificationPreviewByAlertManager({
+  alertManagerSource,
+  potentialInstances,
+  onlyOneAM,
+}: {
+  alertManagerSource: AlertManagerNameWithImage;
+  potentialInstances: Labels[];
+  onlyOneAM: boolean;
+}) {
+  const styles = useStyles2(getStyles);
+
+  const { routesByIdMap, receiversByName, matchingMap, loading, error } = useAlertmanagerNotificationRoutingPreview(
+    alertManagerSource.name,
+    potentialInstances
+  );
+
+  if (error) {
+    return (
+      <Alert title="Cannot load Alertmanager configuration" severity="error">
+        {error.message}
+      </Alert>
+    );
+  }
+
+  if (loading) {
+    return <LoadingPlaceholder text="Loading routing preview..." />;
+  }
+
+  const matchingPoliciesFound = matchingMap.size > 0;
+
+  return matchingPoliciesFound ? (
+    <div className={styles.alertManagerRow}>
+      {!onlyOneAM && (
+        <Stack direction="row" alignItems="center">
+          <div className={styles.firstAlertManagerLine}></div>
+          <div className={styles.alertManagerName}>
+            {' '}
+            Alert manager:
+            <img src={alertManagerSource.img} alt="" className={styles.img} />
+            {alertManagerSource.name}
+          </div>
+          <div className={styles.secondAlertManagerLine}></div>
+        </Stack>
+      )}
+      <Stack gap={1} direction="column">
+        {Array.from(matchingMap.entries()).map(([routeId, instanceMatches]) => {
+          const route = routesByIdMap.get(routeId);
+          const receiver = route?.receiver && receiversByName.get(route.receiver);
+          if (!route || !receiver) {
+            return null;
```

Comment:
Following our Slack conversation we should probably not return `null` here and fail silently – we've made some changes to work with the entire resolved route tree with its inherited properties but in the event it's still null we should `throw` and make sure the matching policies component is wrapped with an error boundary.

Reply:
Additionally we should probably think about how to test this scenario – perhaps we need to add an additional test with a policy that inherits the receiver from the parent.

Reply:
💯

Reply:
Done!